pages/api/props.js

```diff
@@ -0,0 +1,21 @@
+import { getPage } from '../../api/cf';
+
+const isDev = process.env.NODE_ENV === 'development';
+
+export default async function handler(req, res) {
+    if (!isDev) {
+        return res.status(404);
+    }
+
+    const { path } = req.query;
+    
+    if (!path) {
+        return res.status(500).send('Required field `path` is empty.');
+    }
+
+    const sanitizedPath = path.replace(/^\/|\/$/g, '') || '/';
+
+    const pageData = await getPage(sanitizedPath);
+
+    res.status(200).json({ page: pageData });
```

Comment:
the next.js router has two properties
- `asPath`
- `pathname`

https://nextjs.org/docs/api-reference/next/router#router-object

Send them both (don't send `location.pathname`)

Then use pathname to require the correct page file:

```js
const page = require(`../${pathname}.js`)
const pageData = page.getStaticProps({
  params: {
    slug: sanitizedPath.replace(/^\/|\/$/g, '').split('/')
  }
});
```

In other words, don't call `const pageData = await getPage(sanitizedPath);`
Call the `getStaticProps` function directly.
This way you will have single source of truth for computation of page properties.

Reply:
I've fixed all of the above, but didn't manage to load the module using:
`const page = require(`../${pathname}.js`)`

Reply:
What does it say?
Maybe the pathname has leading or trailing slash that needs to be removed. Let’s sync in that. I want to find the best solution that we can then write a guide or a post about